File: Client/src/dtos/userDtos.ts
export interface UsersListItemDto {
  id: string;
  displayName: string;
  userName: string;
}

export interface UpdateProfileDto {
  displayName?: string | null;
  bio?: string | null;
  phoneNumber?: string | null;
}

export interface UserProfileDto {
  id: string;
  userName: string;
  displayName?: string | null;
  bio?: string | null;
  phoneNumber?: string | null;
}

export interface UserDto {
  id: string;
  userName: string;
  displayName?: string | null;
  bio?: string | null;
  avatarId: string;
  isYou: boolean;
  isFollowing: boolean;
  followerCount: number;
  followingCount: number;
}

File: Server/Socialgram/DTOs/UserDtos.cs
namespace Socialgram.DTOs
{
    public record class UsersListItemDto(Guid Id,
                                         string DisplayName,
                                         string UserName);

    public record class UpdateProfileDto(string? DisplayName,
                                         string? Bio,
                                         string? PhoneNumber);

    public record class UserProfileDto(Guid Id,
                                       string UserName,
                                       string? DisplayName,
                                       string? Bio,
                                       string? PhoneNumber);

    public record class UserDto(Guid Id,
                                string UserName,
                                string? DisplayName,
                                string? Bio,
                                Guid AvatarId,
                                bool IsYou,
                                bool IsFollowing,
                                int FollowerCount = 0,
                                int FollowingCount = 0);              
}
